### DISCUSSION TITLE
Measuring events on intel Alder/Raptor Lake with PCM

### DISCUSSION THREAD

**Discussion by mlurbe97** [opcm/pcm#543] [Q&A]
I'm doing measurements with an Intel Alder/Raptor Lake using linux kernel 6.3 and latest intel pcm. 
I'm using the pcm api to program an application manager in C and I want to measure per core events.
But when using the pcm api for C, there is no way to program specific events for each core, and the problem is that P-cores and E-cores dosn't accept the same events. Any solution to this?

Another question is that I would like to program perf events using intel pcm and I'm curious if there is a way to program them using the perf event names like
`perf stat -e cpu_core/topdown-br-mispredict/`
doing the same with pcm
`pcm-core -e cpu_core/topdown-br-mispredict/`
I'm asking this because the intel perfmon events doen't seems to work correctly if i do not specify all the parameters of each event, and probably I'm measuring wrong events due to misspelled event parameters.

If I try
`sudo pcm-core -e name=topdown-br-mispredict`
PCM dosn't say there is an error with the event name and start counting the event, but only 0s appears

We are testing using Ubuntu 22.04 LTS, Linux Kernel 6.3 and the CPU Intel Core i9 13900K

**Comment by ogbrugge** [comment]
Does the answer given in issue #544 help you or do you need further help?
